packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1133,53 +1133,62 @@ class ThemeData with Diagnosticable {
 
   /// A temporary flag used to opt-in to Material 3 features.
   ///
-  /// If true, then components that have been migrated to Material 3 will
+  /// If true, then widgets that have been migrated to Material 3 will
   /// use new colors, typography and other features of Material 3.
   /// If false, they will use the Material 2 look and feel.
   ///
+  /// During the migration to Material 3, turning this on may yield
+  /// inconsistent look and feel in your app as some widgets are migrated
+  /// while others have yet to be.
+  ///
+  /// Defaults to false. When the Material 3 specification is deemed complete
+  /// and all widgets are migrated on stable, we will change this flag to be
+  /// true by default. After that change has landed on stable, we will deprecate
+  /// this flag and remove all uses of it. At that point, the `material` library
+  /// will aim to only support Material 3.
+  ///
+  /// This flag affects styles and components.
+  /// 
+  /// ## Styles
+  ///   * [Color](https://m3.material.io/styles/color/the-color-system): [ColorScheme], [Material]
+  ///   * [Shape](https://m3.material.io/styles/shape) (see components below)
+  ///   * [Typography](https://m3.material.io/styles/typography): `typography` (see table below)
+  ///
+  /// ## Components
+  ///   * [Common buttons](https://m3.material.io/components/buttons): [TextButton], [OutlinedButton], [ElevatedButton]
```

Comment:
Perhaps, differentiate spec links from API doc links?

* Common buttons: [TextButton], [OutlinedButton], [ElevatedButton] ([spec](https://m3.material.io/components/buttons))

Or remove spec link:
* Common buttons: [TextButton], [OutlinedButton], [ElevatedButton]

Reply:
I would opt for leaving the spec link out here. I think most of these have links to the spec in the individual component docs (or should if they don't).

Reply:
I added spec links to all API docs